### PR TITLE
Cleanup `tensorflow/c/experimental/gradients` Part 1

### PR DESCRIPTION
**Pull request by vnghia**
@saxenasaurabh 

One thing I don't understand, if computing numerical gradients with TensorFloat-32 is numerically unstable, does disabling TensorFloat-32 inside `gradient_checker` make a lot more sense than disabling it inside the test, since we will have to disable TF-32 in all binaries depending on `gradient_checker` anyway ?

### CODE REVIEW

**Review by saxenasaurabh — APPROVED** [review]

**Review by saxenasaurabh — APPROVED** [review]

**Comment by saxenasaurabh** [comment]
tensorflow/c/eager/unified_api_test.cc:124:76: error: invalid conversion from 'tensorflow::int64* {aka long long int*}' to 'int64_t* {aka long int*}' [-fpermissive]
         TestTensorHandleWithDimsFloat(ctx.get(), data, dim_sizes, 2, &x_raw);

Please fix.

**Comment by vnvo2409** [comment]
> tensorflow/c/eager/unified_api_test.cc:124:76: error: invalid conversion from 'tensorflow::int64* {aka long long int*}' to 'int64_t* {aka long int*}' [-fpermissive]
TestTensorHandleWithDimsFloat(ctx.get(), data, dim_sizes, 2, &x_raw);

Done

**Review by saxenasaurabh — APPROVED** [review]

File: tensorflow/c/eager/gradient_checker_test.cc [hunk 1]
```diff
@@ -15,28 +15,66 @@ limitations under the License.
...
+  ASSERT_EQ(num_elem_numerical, num_grad);
+
+  float* dnumerical = new float[num_elem_numerical]{0};
+  memcpy(&dnumerical[0], TF_TensorData(numerical_tensor),
+         TF_TensorByteSize(numerical_tensor));
+
+  for (int j = 0; j < num_grad; j++) {
+    ASSERT_NEAR(dnumerical[j], expected_grad[j], abs_error);
+  }
+  delete dnumerical;
```
Comment (saxenasaurabh) [inline tensorflow/c/eager/gradient_checker_test.cc:55]: Use `delete[]`

Comment (vnvo2409) [reply tensorflow/c/eager/gradient_checker_test.cc:55]: Done

**Review by saxenasaurabh — APPROVED** [review]

**Comment by rthadur** [comment]
@vnvo2409 can you please resolve conflicts ?

**Comment by vnvo2409** [comment]
> @vnvo2409 can you please resolve conflicts ?

Done

**Review by saxenasaurabh — APPROVED** [review]

**Comment by saxenasaurabh** [comment]
Looks like `eager/BUILD` has conflicts. Rebase maybe? Also `nn_grad_test` is segfaulting internally. I will try to patch and see what's going on.

**Comment by saxenasaurabh** [comment]
Ah got it. Let's fix the SoftmaxModel. The leaks we can address in a follow-up.

**Comment by vnvo2409** [comment]
> Let's fix the SoftmaxModel.

Done

**Review by saxenasaurabh — APPROVED** [review]

**Comment by vnvo2409** [comment]
@saxenasaurabh 
The copybara/feedback failed
